haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -434,11 +438,93 @@ def save(self, save_dir: Union[Path, str]):
         raise NotImplementedError(f"Saving is not implemented for {self.__class__}")
 
 
+class _BedrockEmbeddingEncoder(_BaseEmbeddingEncoder):
+    def __init__(self, retriever: "EmbeddingRetriever"):
+        boto3_import.check()
+
+        # See https://docs.aws.amazon.com/bedrock/latest/userguide/embeddings.html for more details
+        # The maximum input text is 8K tokens and the maximum output vector length is 1536
+        # Bedrock Titan embeddings do not support batch operations
+        self.aws_config = retriever.aws_config
+        self.client = self.initialize_boto3_session().client("bedrock-runtime")
+        self.model: str = next(
+            (m for m in BEDROCK_EMBEDDING_MODELS if m in retriever.embedding_model), "amazon.titan-embed-text-v1"
+        )
+
+    def initialize_boto3_session(self):
+        if self.aws_config:
+            profile_name = self.aws_config.get("profile_name", None)
+            access_key = self.aws_config.get("aws_access_key_id", None)
+            secret_key = self.aws_config.get("aws_secret_access_key", None)
+            region = self.aws_config.get("region_name", None)
+            if profile_name:
+                try:
+                    return boto3.Session(profile_name=profile_name, region_name=region)
+                except Exception as e:
+                    raise ValueError(f"AWS client error {e}")
+            elif access_key and secret_key:
+                return boto3.Session(aws_access_key_id=access_key, aws_secret_access_key=secret_key, region_name=region)
+        else:
+            try:
+                return boto3.Session()
+            except Exception as e:
+                raise ValueError(f"AWS client error {e}")
```

Comment:
Is this handling of different cases really necessary?
In https://github.com/deepset-ai/haystack/blob/f998bf4a4fc02b9f4d847bef065bb3d1666fb2cf/haystack/nodes/prompt/invocation_layer/aws_base.py#L68-L74 we just pass all params.